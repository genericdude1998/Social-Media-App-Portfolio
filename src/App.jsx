import React from 'react';
import { Route, BrowserRouter, Routes } from 'react-router-dom';
import ConnectedLogin from './view/Login/Login';
import ConnectedFeedWithTheme from './view/Feed/Feed';
import ConnectedUser from './view/UsersInfo/User';
import NavBar from './view/Feed/NavBar';
import ConnectedNPC from './view/Feed/NewPost/NewPostCreator';
import ConnectedThemeProvider from './view/ThemeProvider';

const App = () => {
    return (
        <div>
            <ConnectedThemeProvider>
                <BrowserRouter>
                    <Routes>
                        <Route path='/' element={<ConnectedLogin />}/>
                        <Route path='/feed' element=
                            {
                                <>
                                    <NavBar/>
                                    <ConnectedFeedWithTheme/>
                                </>
                            }  
                        />
                        <Route path='/userInfo/:id' element=
                            {
                                <>
                                    <NavBar/>
                                    <ConnectedUser/>
                                </>
                            }/>
                        <Route path='/createPost' element={
                            <>
                                <NavBar/>
                                <ConnectedNPC/>
                            </>
                        }/>
                    </Routes>
                </BrowserRouter>
            </ConnectedThemeProvider>
        </div>
    );
}

export default App;
