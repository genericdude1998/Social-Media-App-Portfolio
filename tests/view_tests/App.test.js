import React from 'react'
import { shallow } from 'enzyme';
import App from '../../src/App';

let wrapper = shallow(<App/>);

describe('App', () => {
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});