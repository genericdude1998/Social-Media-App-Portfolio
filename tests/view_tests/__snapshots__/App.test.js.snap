// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <BrowserRouter>
    <Routes>
      <Route
        element={<Memo(Connect(Login)) />}
        path="/"
      />
      <Route
        element={
          <React.Fragment>
            <NavBar />
            <Memo(Connect(Feed)) />
          </React.Fragment>
        }
        path="/feed"
      />
      <Route
        element={
          <React.Fragment>
            <NavBar />
            <Memo(Connect(User)) />
          </React.Fragment>
        }
        path="/userInfo/:id"
      />
    </Routes>
  </BrowserRouter>
</div>
`;
